source/common/http/hash_policy.cc

```diff
@@ -39,14 +42,41 @@ class HeaderHashMethod : public HashMethodImplBase {
                                     const StreamInfo::FilterStateSharedPtr) const override {
     absl::optional<uint64_t> hash;
 
-    // TODO(mattklein123): Potentially hash on all headers.
     const auto header = headers.get(header_name_);
+    bool check_multiple_values =
+        Runtime::runtimeFeatureEnabled("envoy.reloadable_features.hash_multiple_header_values");
```

Comment:
oops, this line should be inside the '!header.empty()' check block, will update next round.